optional_plugins/robot/avocado_robot/__init__.py

```diff
@@ -0,0 +1,132 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2017
+# Authors: Amador Pahim <apahim@redhat.com>
+
+import collections
+import logging
+import subprocess
+
+from StringIO import StringIO
+
+from avocado.core import exceptions
+from avocado.core import loader
+from avocado.core import output
+from avocado.core import test
+from avocado.core.plugin_interfaces import CLI
+from avocado.core.runner import TestRunner
+from avocado.utils import path as utils_path
+from robot import run
+from robot.parsing.model import TestData
+
+
+class RobotTest(test.SimpleTest):
+
+    """
+    Run a Robot command as a SIMPLE test.
+    """
+
+    def __init__(self,
+                 name,
+                 params=None,
+                 base_logdir=None,
+                 job=None,
+                 external_runner=None):
+        super(RobotTest, self).__init__(name, params, base_logdir, job)
+
+    @property
+    def filename(self):
+        """
+        Returns the path of the robot test suite.
+        """
+        return self.name.name.split(':')[0]
+
+    def test(self):
+        """
+        Create the Robot command and execute it.
+        """
+        suite_name, test_name = self.name.name.split(':')[1].split('.')
+        log_stdout = output.LoggingFile(logger=[self.log], level=logging.INFO)
+        log_stderr = output.LoggingFile(logger=[self.log], level=logging.ERROR)
+        result = run(self.filename,
+                     suite=suite_name,
+                     test=test_name,
+                     outputdir=self.outputdir,
+                     stdout=log_stdout,
+                     stderr=log_stderr)
+        if result:
+            self.fail('Robot execution returned a '
+                      'non-0 exit code (%s)' % result)
+
+
+class RobotLoader(loader.TestLoader):
+    """
+    Robot loader class
+    """
+    name = "robot"
+
+    def __init__(self, args, extra_params):
+        super(RobotLoader, self).__init__(args, extra_params)
+
+    def discover(self, url, which_tests=False):
+        avocado_suite = []
+        robot_suite = self._find_tests(TestData(parent=None, source=url),
```

Comment:
So to actually execute `robot` tests you have to run `avocado run --loaders robot --robot $PATH` otherwise due the `etc/avocado/avocado.conf` default which is `loaders = ['file', '@DEFAULT']` the `file` loader has higher priority and is always present.

How about making `robot` always registered and ask people to use `--loaders` as a standard mechanism to select the loader?